{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "67527faf-ed40-40c3-8747-70f680f36744",
   "metadata": {},
   "source": [
    "<strong><h1>Sequenced (twice) - pie charts </h1></strong>\r\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "765be4b9-4f31-4e7a-9fe6-f13f952162ca",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "excel_file_path1 = r'C:\\Users\\hayae\\Desktop\\New_opencravat_summary_sheets.xlsx'\n",
    "excel_file_path2 = r'C:\\Users\\hayae\\Desktop\\New_opencravat_summary_sheets.xlsx'\n",
    "\n",
    "# Sheet names for the two Excel files\n",
    "sheet_name1 = 'PR_080'\n",
    "sheet_name2 = 'PR_080_v1'\n",
    "\n",
    "df1 = pd.read_excel(excel_file_path1, sheet_name=sheet_name1)\n",
    "df2 = pd.read_excel(excel_file_path2, sheet_name=sheet_name2)\n",
    "\n",
    "# Filter the desired columns\n",
    "columns_to_match = ['Chrom', 'Gene', 'Position', 'Ref_Base','Alt_Base', 'Sequence_Ontology']\n",
    "df1 = df1[columns_to_match]\n",
    "df2 = df2[columns_to_match]\n",
    "\n",
    "# Merge dataframes on the columns to find matching rows\n",
    "merged_df = pd.merge(df1, df2, how='inner', on=columns_to_match)\n",
    "\n",
    "# Calculate percentages and counts\n",
    "total_rows_df1 = len(df1)\n",
    "matching_rows = len(merged_df)\n",
    "non_matching_rows = total_rows_df1 - matching_rows\n",
    "\n",
    "# Confirm that matching_rows is not greater than total_rows_df1\n",
    "if matching_rows > total_rows_df1:\n",
    "    print(\"Warning: More matching rows than total rows in df1.\")\n",
    "    matching_rows = total_rows_df1\n",
    "\n",
    "percentage_matches = (matching_rows / total_rows_df1) * 100 if total_rows_df1 > 0 else 0import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "import matplotlib\n",
    "\n",
    "matplotlib.rcParams['font.family'] = 'Times New Roman'\n",
    "\n",
    "\n",
    "excel_file_path1 = r'C:\\Users\\hayae\\Desktop\\New_opencravat_summary.xlsx'\n",
    "excel_file_path2 = r'C:\\Users\\hayae\\Desktop\\New_opencravat_summary.xlsx'\n",
    "\n",
    "sheet_name1 = 'PR_081'\n",
    "sheet_name2 = 'PR_081_v1'\n",
    "\n",
    "\n",
    "df1 = pd.read_excel(excel_file_path1, sheet_name=sheet_name1)\n",
    "df2 = pd.read_excel(excel_file_path2, sheet_name=sheet_name2)\n",
    "\n",
    "# Filter the desired columns\n",
    "columns_to_match_filter = ['Alternate_reads', 'Total_reads', 'Variant_AF']\n",
    "columns_to_match_merge = ['Chrom', 'Gene', 'Position', 'Ref_Base','Alt_Base', 'Sequence_Ontology']\n",
    "df1_filtered = df1[(df1['Alternate_reads'] >= 5) & (df1['Total_reads'] >= 500) & (df1['Variant_AF'] >= 0.005)]\n",
    "df2_filtered = df2[(df2['Alternate_reads'] >= 5) & (df2['Total_reads'] >= 500) & (df2['Variant_AF'] >= 0.005)]\n",
    "\n",
    "# Merge dataframes on the columns to find matching rows\n",
    "merged_df = pd.merge(df1_filtered, df2_filtered, how='inner', on=columns_to_match_merge)\n",
    "\n",
    "# Calculate percentages and counts\n",
    "total_rows_df1 = len(df1_filtered)\n",
    "matching_rows = len(merged_df)\n",
    "non_matching_rows = total_rows_df1 - matching_rows\n",
    "\n",
    "# Confirm that matching_rows is not greater than total_rows_df1\n",
    "if matching_rows > total_rows_df1:\n",
    "    print(\"Warning: More matching rows than total rows in df1.\")\n",
    "    matching_rows = total_rows_df1\n",
    "\n",
    "percentage_matches = (matching_rows / total_rows_df1) * 100 if total_rows_df1 > 0 else 0\n",
    "percentage_non_matches = max(0, 100 - percentage_matches)  # Ensure this is non-negative\n",
    "\n",
    "# Plot pie chart\n",
    "labels = ['Matches', 'Non-Matches']\n",
    "sizes = [matching_rows, non_matching_rows]\n",
    "colors = [\"#56B4E9\", 'lightcoral']\n",
    "\n",
    "fig, ax = plt.subplots(figsize=(8, 8))  # Adjust the figsize here\n",
    "patches, texts, autotexts = ax.pie(sizes, labels=None, colors=colors, autopct='%1.1f%%', startangle=360, textprops={'fontsize': 14})  # Set fontsize for autotexts\n",
    "\n",
    "ax.axis('equal')\n",
    "ax.set_title('Comparing Variants of Patient 081 V1 & V2 After Filtering' , fontweight='bold', fontsize = 16)\n",
    "\n",
    "# Get the position of the autotexts\n",
    "autotexts_pos = [autotext.get_position() for autotext in autotexts]\n",
    "\n",
    "# Add annotations for Matches and Non-Matches labels next to percentages\n",
    "matches_x, matches_y = autotexts_pos[0]\n",
    "non_matches_x, non_matches_y = autotexts_pos[1]\n",
    "\n",
    "# Add count and percentage labels only once\n",
    "matches_label = f\"Matches: {matching_rows}\"\n",
    "non_matches_label = f\"Non-Matches: {non_matching_rows}\"\n",
    "\n",
    "# Move the matches label downwards\n",
    "ax.annotate(matches_label, xy=(matches_x, matches_y), xytext=(matches_x, matches_y - 0.3),\n",
    "             fontsize=14, ha='center')\n",
    "\n",
    "# Move the non-matches label downwards\n",
    "ax.annotate(non_matches_label, xy=(non_matches_x, non_matches_y), xytext=(non_matches_x, non_matches_y - 0.3),\n",
    "             fontsize=14, ha='center')\n",
    "\n",
    "# Adding text for filtering criteria\n",
    "filter_text = 'Filtering:\\n - Alternate Reads ≥ 5\\n- Total Reads ≥ 500\\n- Variant Allele Frequency ≥ 0.005'\n",
    "ax.text(1.4, 0.4, filter_text, fontsize=12, ha='left', va='center')\n",
    "\n",
    "# Save the plot as a PNG file\n",
    "plt.savefig(r'C:\\Users\\hayae\\Desktop\\matching_rows_pr_081_v1_v2_filtering.png', bbox_inches='tight')\n",
    "\n",
    "plt.show()\n",
    "\n",
    "percentage_non_matches = max(0, 100 - percentage_matches)  # Ensure this is non-negative\n",
    "\n",
    "# Plot pie chart\n",
    "labels = ['Matches', 'Non-Matches']\n",
    "sizes = [matching_rows, non_matching_rows]\n",
    "colors = [\"#56B4E9\", 'lightcoral']\n",
    "\n",
    "fig, ax = plt.subplots()\n",
    "patches, texts, autotexts = ax.pie(sizes, labels=labels, colors=colors, autopct='%1.1f%%', startangle=45)\n",
    "ax.axis('equal')\n",
    "ax.set_title('Percentage of Matching Rows PR 001 V1 & V2')\n",
    "\n",
    "# Add count labels\n",
    "for i, text in enumerate(texts):\n",
    "    percentage = sizes[i] / total_rows_df1 * 100 if total_rows_df1 > 0 else 0\n",
    "    text.set_text(f\"{sizes[i]} ({percentage:.1f}%)\")\n",
    "\n",
    "# Save the plot as a PNG file\n",
    "plt.savefig(r'C:\\Users\\hayae\\Desktop\\matching_rows_pr_001_v1_v2.png')\n",
    "\n",
    "plt.show()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3ad16e4e-68e6-4786-81ba-c90c15d9f3e5",
   "metadata": {},
   "source": [
    "<strong><h1>Finding all AR genes</h1></strong>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2dd8a6fb-fa4a-4c91-834d-b426790f27d8",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Define the Excel file path\n",
    "excel_file_path = r'C:\\Users\\hayae\\Desktop\\New_opencravat_summary_sheets.xlsx'\n",
    "\n",
    "# Define a list of sheet names to read\n",
    "sheets_to_read = ['PR_001_new', 'PR_014_new', 'PR_015_new', 'PR_016_new', 'PR_017_new', 'PR_018_new', \n",
    "                  'PR_020_new', 'PR_021_new', 'PR_122_new', 'PR_126_new', 'PR_127_new', 'PR_128_new', \n",
    "                  'PR_133_new', 'PR_135_new', 'PR_138_new', 'PR_142_new', 'PR_175_new', 'PR_189_new', \n",
    "                  'PR_200_new', 'PR_201_new', 'PR_209_new', 'PR_245_new', 'PR_270_new', 'PR_282_new', \n",
    "                  'PR_283_new', 'PR_298_1_new', 'PR_28', 'PR_35', 'PR_36', 'PR_37', 'PR_38', \n",
    "                  'PR_39', 'PR_40', 'PR_41', 'PR_42', 'PR_43', 'PR_44', 'PR_45', 'PR_46', 'PR_48', \n",
    "                  'PR_49', 'PR_50', 'PR_52', 'PR_53', 'PR_54', 'PR_55', 'PR_56', 'PR_57', \n",
    "                  'PR_58', 'PR_59', 'PR_068', 'PR_070', 'PR_080', 'PR_081', 'PR_094', 'PR_095', \n",
    "                  'PR_096', 'PR_001_v2', 'PR_015_v2', 'PR_023_v2', 'PR_064_v2', 'PR_066_v2', 'PR_067_v2',\n",
    "                  'PR_068_v1', 'PR_071', 'PR_072', 'PR_077', 'PR_080_v1', 'PR_081_v1', 'PR_109', 'PR_110', 'PR_112',\n",
    "                  'PR_113', 'PR_115', 'PR_117', 'PR_119', 'PR_120', 'PR_121', 'PR_123', 'PR_124', 'PR_125', 'PR_126', \n",
    "                  'PR_127', 'PR_128']\n",
    "\n",
    "# Genes to search for\n",
    "genes_to_search = [\"ARID1A\", \"ATM\", \"ATR\", \"ATRX\", \"BARD1\", \"BRCA1\", \"BRCA2\", \"BRIP1\", \"CDK12\", \n",
    "                   \"CHEK1\", \"CHEK2\", \"FANCA\", \"FANCF\", \"FANCG\", \"FANCI\", \"FANCL\", \"FANCM\", \n",
    "                   \"HDAC2\", \"MLH1\", \"MRE11A\", \"MSH2\", \"NBN\", \"PALB2\", \"PALP2\", \"PPP2R2A\", \"RAD50\", \n",
    "                   \"RAD51\", \"RAD51B\", \"RAD51C\", \"RAD51D\", \"RAD54\", \"RAD54L\", \"WRN\"]\n",
    "\n",
    "# Variants to exclude\n",
    "variants_to_exclude = [\n",
    "    {\"Gene\": \"ATRX\", \"cDNA_change\": \"c.2785G>C\"},\n",
    "    {\"Gene\": \"BARD1\", \"cDNA_change\": \"c.1518_1519delinsCA\"},\n",
    "    {\"Gene\": \"FANCA\", \"cDNA_change\": \"c.2574C>G\"}\n",
    "]\n",
    "\n",
    "# Create an empty DataFrame to hold the final results\n",
    "final_result = pd.DataFrame()\n",
    "\n",
    "# Iterate over each sheet to read\n",
    "for sheet_name in sheets_to_read:\n",
    "    # Read the Excel file\n",
    "    df = pd.read_excel(excel_file_path, sheet_name=sheet_name)\n",
    "    \n",
    "    # Apply the initial filters\n",
    "    df_filtered = df[(df['Alternate_reads'] >= 5) & (df['Total_reads'] >= 500) & (df['Variant_AF'] >= 0.005)]\n",
    "    \n",
    "    # Exclude specific variants\n",
    "    for variant in variants_to_exclude:\n",
    "        df_filtered = df_filtered[~((df_filtered['Gene'] == variant['Gene']) & (df_filtered['cDNA_change'] == variant['cDNA_change']))]\n",
    "    \n",
    "    # Create a dictionary to hold gene presence (1) or absence (0)\n",
    "    gene_presence = {}\n",
    "    \n",
    "    # Iterate over genes to search for\n",
    "    for gene in genes_to_search:\n",
    "        # Check if the gene exists in the 'Gene' column\n",
    "        if gene in df_filtered['Gene'].values:\n",
    "            gene_presence[gene] = 1\n",
    "        else:\n",
    "            gene_presence[gene] = 0\n",
    "            \n",
    "    # Convert dictionary to DataFrame\n",
    "    result_df = pd.DataFrame(gene_presence.items(), columns=['Gene', sheet_name])\n",
    "    \n",
    "    # Set the Gene column as the index\n",
    "    result_df.set_index('Gene', inplace=True)\n",
    "    \n",
    "    # Transpose the DataFrame to have genes as columns and sheet names as rows\n",
    "    result_df = result_df.T\n",
    "    \n",
    "    # Append the transposed result DataFrame to the final result DataFrame\n",
    "    final_result = pd.concat([final_result, result_df])\n",
    "\n",
    "# Reset the index to include the column with sheet names\n",
    "final_result.reset_index(inplace=True)\n",
    "final_result.rename(columns={'index': 'Sheet'}, inplace=True)\n",
    "\n",
    "# Print the final result\n",
    "print(final_result)\n",
    "\n",
    "# Save the final result to a new Excel file\n",
    "final_result.to_excel(r'C:\\Users\\hayae\\Desktop\\AR_gene_presence_all_sheets.xlsx', index=False)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c9d29df6-a1db-439b-b4c3-b24d567a8d83",
   "metadata": {},
   "source": [
    "<strong><h1>Metastasis Presence by Patient</h1></strong>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4852260e-775f-42e6-8e1a-ef18172db92b",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "# Read the Excel file\n",
    "df = pd.read_excel(\"C:/Users/hayae/Desktop/Onco_test.xlsx\", sheet_name=\"Sheet4\")\n",
    "\n",
    "# Set 'Pat ID' as the index\n",
    "df.set_index('Pat ID', inplace=True)\n",
    "\n",
    "# Convert metastasis data to binary indicators (0 for absence, 1 for presence)\n",
    "df_binary = df.notna().astype(int)\n",
    "\n",
    "# Plotting\n",
    "fig, ax = plt.subplots(figsize=(16, 26))  # Adjust figure size as needed (increase height)\n",
    "df_binary.plot(kind='barh', stacked=True, legend='reverse', ax=ax)\n",
    "\n",
    "# Customize y-axis ticks\n",
    "tick_spacing = 1  # Set the spacing between ticks\n",
    "ax.set_yticks(range(0, len(df_binary.index), tick_spacing))  # Set ticks every tick_spacing indices\n",
    "ax.set_yticklabels(df_binary.index[::tick_spacing])  # Set labels to correspond to the new ticks\n",
    "\n",
    "plt.xlabel('Metastasis Type' , fontsize= 15)\n",
    "plt.ylabel('Patient ID' , fontsize= 15)\n",
    "plt.title('Metastasis Presence by Patient', fontsize= 20)\n",
    "plt.legend(title='Metastasis', bbox_to_anchor=(0.79, 1), loc='upper left')\n",
    "\n",
    "# Save the plot\n",
    "plt.savefig('C:/Users/hayae/Desktop/Metastasis Presence by Patient.png')\n",
    "\n",
    "plt.show()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ed041076-52ca-426c-abe1-83827ba53dda",
   "metadata": {},
   "source": [
    "<strong><h1>Amplification and Hotspot Mutations of AR</h1></strong>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e97a5712-0d27-4f7f-93a2-22a71c542f0c",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "# Read the Excel file into a DataFrame\n",
    "df = pd.read_excel(\"C:/Users/hayae/Desktop/Onco_test.xlsx\", sheet_name=\"AR\")\n",
    "\n",
    "# Ensure both 'Yes' and 'No' categories are present even if all entries are 'Yes'\n",
    "high_low_volume_counts = df['high volume, low volume'].value_counts().reindex(['Yes', 'No'], fill_value=0)\n",
    "high_low_risk_counts = df['high risk, low risk'].value_counts().reindex(['Yes', 'No'], fill_value=0)\n",
    "\n",
    "# Calculate percentages\n",
    "high_low_volume_percentages = high_low_volume_counts / high_low_volume_counts.sum() * 100\n",
    "high_low_risk_percentages = high_low_risk_counts / high_low_risk_counts.sum() * 100\n",
    "\n",
    "# Create a bar plot\n",
    "plt.figure(figsize=(16, 10))\n",
    "\n",
    "# Plot for 'high volume, low volume'\n",
    "plt.subplot(1, 2, 1)\n",
    "high_low_volume_counts.plot(kind='bar', color='#E69F00')\n",
    "for index, value in enumerate(high_low_volume_counts):\n",
    "    if high_low_volume_percentages[index] == 0:\n",
    "        plt.text(index, value + 0.5, f'n={value} ({high_low_volume_percentages[index]:.1f}%)', ha='center', fontsize=12)\n",
    "    else:\n",
    "        plt.text(index, value - 0.5, f'n={value} ({high_low_volume_percentages[index]:.1f}%)', ha='center', fontsize=12)\n",
    "plt.title('High Volume vs. Low Volume', fontsize=18)\n",
    "plt.xlabel('Volume', fontsize=12)\n",
    "plt.ylabel('Count', fontsize=12)\n",
    "plt.tick_params(axis='both', which='major', labelsize=12)\n",
    "\n",
    "# Plot for 'high risk, low risk'\n",
    "plt.subplot(1, 2, 2)\n",
    "high_low_risk_counts.plot(kind='bar', color='#56B4E9')\n",
    "for index, value in enumerate(high_low_risk_counts):\n",
    "    if high_low_risk_percentages[index] == 0:\n",
    "        plt.text(index, value + 0.5, f'n={value} ({high_low_risk_percentages[index]:.1f}%)', ha='center', fontsize=12)\n",
    "    else:\n",
    "        plt.text(index, value - 0.5, f'n={value} ({high_low_risk_percentages[index]:.1f}%)', ha='center', fontsize=12)\n",
    "plt.title('High Risk vs. Low Risk', fontsize=18)\n",
    "plt.xlabel('Risk', fontsize=12)\n",
    "plt.ylabel('Count', fontsize=12)\n",
    "plt.tick_params(axis='both', which='major', labelsize=12)\n",
    "\n",
    "# Set main title\n",
    "plt.suptitle('Amplification and Hotspot Mutations of AR', fontsize=20, fontweight='bold')\n",
    "\n",
    "plt.tight_layout()\n",
    "\n",
    "# Save the plot as PNG\n",
    "plt.savefig(\"C:/Users/hayae/Desktop/AR_high_volume_low_volume_vs_high_risk_low_risk.png\")\n",
    "\n",
    "plt.show()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3e09bad2-4995-4e3c-b43f-8b8b0a79f720",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "# Read the Excel file into a DataFrame\n",
    "df = pd.read_excel(\"C:/Users/hayae/Desktop/Onco_test.xlsx\", sheet_name=\"HRR\")\n",
    "\n",
    "# Ensure both 'High' and 'Low' categories are present even if all entries are 'High'\n",
    "high_low_volume_counts = df['high volume, low volume'].value_counts().reindex(['High', 'Low'], fill_value=0)\n",
    "high_low_risk_counts = df['high risk, low risk'].value_counts().reindex(['High', 'Low'], fill_value=0)\n",
    "\n",
    "# Total number of patients\n",
    "total_patients = len(df)\n",
    "\n",
    "# Calculate ATRX counts for each category\n",
    "ATRX_counts_volume = df.groupby('high volume, low volume')['ATRX'].sum()\n",
    "ATRX_counts_risk = df.groupby('high risk, low risk')['ATRX'].sum()\n",
    "\n",
    "# Calculate percentages\n",
    "ATRX_percentages_volume = ATRX_counts_volume / total_patients * 100\n",
    "ATRX_percentages_risk = ATRX_counts_risk / total_patients * 100\n",
    "\n",
    "# Create a bar plot for 'high volume, low volume'\n",
    "plt.figure(figsize=(22, 16))\n",
    "plt.subplot(1, 2, 1)\n",
    "ATRX_counts_volume.plot(kind='bar', color='#E69F00')\n",
    "for index, value in enumerate(ATRX_counts_volume):\n",
    "    bar_width = 0.35\n",
    "    x_coord = index - bar_width / 2  # Adjust x-coordinate to center the label\n",
    "    plt.text(x_coord, value / 2, f'n={value} ({ATRX_percentages_volume[index]:.1f}%)', ha='left', va='center', fontsize=16)\n",
    "plt.title('ATRX Mutation vs. High Volume vs. Low Volume', fontsize=20)\n",
    "plt.xlabel('Volume', fontsize=16)\n",
    "plt.ylabel('Count of ATRX Mutations', fontsize=16)\n",
    "plt.xticks(range(len(ATRX_counts_volume.index)), ATRX_counts_volume.index, rotation=0, fontsize=16)\n",
    "plt.tick_params(axis='both', which='major', labelsize=16)\n",
    "\n",
    "# Create a bar plot for 'high risk, low risk'\n",
    "plt.subplot(1, 2, 2)\n",
    "ATRX_counts_risk.plot(kind='bar', color='#56B4E9')\n",
    "for index, value in enumerate(ATRX_counts_risk):\n",
    "    bar_width = 0.35\n",
    "    x_coord = index - bar_width / 2  # Adjust x-coordinate to center the label\n",
    "    plt.text(x_coord, value / 2, f'n={value} ({ATRX_percentages_risk[index]:.1f}%)', ha='left', va='center', fontsize=16)\n",
    "plt.title('ATRX Mutation vs. High Risk vs. Low Risk', fontsize=20)\n",
    "plt.xlabel('Risk', fontsize=16)\n",
    "plt.ylabel('Count of ATRX Mutations', fontsize=16)\n",
    "plt.xticks(range(len(ATRX_counts_risk.index)), ATRX_counts_risk.index, rotation=0, fontsize=16)\n",
    "plt.tick_params(axis='both', which='major', labelsize=16)\n",
    "\n",
    "plt.suptitle('ATRX Mutations for all Patients', fontsize=30, fontweight='bold')\n",
    "\n",
    "# Save the plot as PNG\n",
    "plt.savefig(\"C:/Users/hayae/Desktop/_high_volume_low_volume_vs_high_risk_low_risk.png\")\n",
    "\n",
    "plt.tight_layout()\n",
    "plt.show()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "31051fd0-1c17-4c10-b912-b6ed4f60577c",
   "metadata": {},
   "source": [
    "<strong><h1>Total HRR Gene Variants Across Patients</h1></strong>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bfb63a85-14c5-4570-8322-98b2e7c8892c",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "# Read the Excel file into a DataFrame\n",
    "df = pd.read_excel(\"C:/Users/hayae/Desktop/Onco_test.xlsx\", sheet_name=\"HRR\")  # Update sheet name if necessary\n",
    "\n",
    "# Define the list of genes\n",
    "genes = ['ARID1A', 'ATM', 'ATR', 'ATRX', 'BARD1', 'BRCA1', 'BRCA2', 'BRIP1', 'CDK12', 'CHEK1',\n",
    "         'CHEK2', 'FANCA', 'FANCF', 'FANCG', 'FANCI', 'FANCL', 'FANCM', 'HDAC2', 'MLH1', 'MRE11A',\n",
    "         'MSH2', 'NBN', 'PALB2', 'PALP2', 'PPP2R2A', 'RAD50', 'RAD51', 'RAD51B', 'RAD51C', 'RAD51D',\n",
    "         'RAD54', 'RAD54L', 'WRN']\n",
    "\n",
    "# Sum the occurrences of each gene for all patients\n",
    "gene_counts = df[genes].sum(axis=1)\n",
    "\n",
    "# Sort patients by the total number of gene variants\n",
    "sorted_indices = gene_counts.argsort()\n",
    "sorted_gene_counts = gene_counts.iloc[sorted_indices]\n",
    "sorted_patients = df.iloc[sorted_indices, 0]\n",
    "\n",
    "# Define patient IDs to be highlighted\n",
    "highlighted_LPC = ['PR_016', 'PR_018', 'PR_020', 'PR_021', 'PR_270', 'PR_282', 'PR_283', 'PR_46', 'PR_48', 'PR_49']\n",
    "highlighted_CRPC = ['PR_001', 'PR_015', 'PR_017', 'PR_122', 'PR_135', 'PR_142', \n",
    "                        'PR_189', 'PR_201', 'PR_209', 'PR_245', 'PR_36', 'PR_42', \n",
    "                        'PR_54', 'PR_56']\n",
    "\n",
    "# Plotting\n",
    "plt.figure(figsize=(20, 10))\n",
    "\n",
    "# Plotting all patients\n",
    "plt.bar(sorted_patients, sorted_gene_counts, color='#56B4E9', label='Gene Variants')\n",
    "\n",
    "# Plotting highlighted patients\n",
    "for patient_id in highlighted_LPC:\n",
    "    index = df[df.iloc[:, 0] == patient_id].index[0]\n",
    "    plt.bar(df.iloc[index, 0], gene_counts[index], color='#E69F00')\n",
    "\n",
    "    for patient_id in highlighted_CRPC:\n",
    "        index = df[df.iloc[:, 0] == patient_id].index[0]\n",
    "        plt.bar(df.iloc[index, 0], gene_counts[index], color=\"#FF9999\")\n",
    "\n",
    "plt.title('Total HRR Gene Variants Across Patients', fontsize=22, fontweight='bold')\n",
    "plt.xlabel('Patient ID', fontsize=18)\n",
    "plt.ylabel('Total Gene Variants', fontsize=18)\n",
    "plt.xticks(rotation=90, fontsize=16)  # Rotate x-axis labels and increase font size\n",
    "plt.yticks(fontsize=16)\n",
    "# Create legend\n",
    "plt.legend(['HSPC', 'LPC', 'CRPC'], fontsize=14)\n",
    "\n",
    "plt.tight_layout()\n",
    "\n",
    "# Save the plot as PNG\n",
    "plt.savefig(\"C:/Users/hayae/Desktop/sorted_Total_HRR_gene_variants_per_patient.png\")\n",
    "\n",
    "plt.show()\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
